kratos/input_output/logger.cpp

```diff
@@ -49,6 +50,20 @@ namespace Kratos
     }
   }
 
+  void Logger::RemoveOutput(LoggerOutput::Pointer pTheOutput)
+  {
+    KRATOS_TRY
+
+    #pragma omp critical
+    {
+      auto i = std::find(GetOutputsInstance().begin(), GetOutputsInstance().end(), pTheOutput);
+      KRATOS_ERROR_IF(i == GetOutputsInstance().end()) << pTheOutput->Info() << " was not found in the list of loggers.\n";
+      GetOutputsInstance().erase(i);
+    }
+
+    KRATOS_CATCH("");
```

Comment:
like this if two thread call this the second one to enter will always crash!

Reply:
Not exactly. It depends. First I was thinking to throw an error if this is called in a parallel region, then I thought of the following block.

https://github.com/KratosMultiphysics/Kratos/blob/77633dd7fa7ad6f55bbb1f75ae847666b4a6b435/kratos/input_output/logger.cpp#L44-L50

If the addition is also called in OMP, then it will add the same `Logger` `N` times, so I think then we need to remove it `N` times as well. (I am not sure what is the logic in calling this in OMP loop in first place.)

Reply:
Maybe @pooyan-dadvand knows the reason for putting the OMP block here? Is it gonna be called in OMP parallel regions?

Reply:
why not add a small check and do nothing in the second enter?

Reply:
done :)

Reply:
It is simply an error! @sunethwarna, thanks for spotting it. 

IMO it should check if it is called from a parallel region and send an error.